File: src/main/java/net/hectus/neobb/turn/default_game/mob/TPufferfish.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.WaterClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.entity.PufferFish;

import java.util.List;

public class TPufferfish extends MobTurn<PufferFish> implements BuffFunction, WaterClazz {
    public TPufferfish(NeoPlayer player) { super(player); }
    public TPufferfish(PufferFish data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 6;
    }

    @Override
    public void apply() {
        // TODO: Make player poisoned for 2 turns and kill him afterwards, if not healed.
    }

    @Override
    public List<Buff> buffs() {
        return List.of();
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/mob/TPufferfish.kt
package net.hectus.neobb.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.WaterClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import org.bukkit.entity.PufferFish
import org.bukkit.potion.PotionEffectType

class TPufferfish(data: PufferFish?, cord: Cord?, player: NeoPlayer?) : MobTurn<PufferFish>(data, cord, player), BuffFunction, WaterClazz {
    override val cost: Int = 6

    override fun apply() {
        player!!.game.turnScheduler.runTaskLater(ScheduleID.POISON, 2) {
            if (player.nextPlayer().player.hasPotionEffect(PotionEffectType.POISON))
                player.game.eliminate(player.nextPlayer())
        }
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(Effect(PotionEffectType.POISON, target = Buff.BuffTarget.NEXT))
    }
}
